src/pipecat/processors/aggregators/openai_llm_context.py

```diff
@@ -112,11 +112,59 @@ def get_messages_for_logging(self) -> str:
             msgs.append(msg)
         return json.dumps(msgs)
 
-    def from_standard_message(self, message):
+    def from_standard_message(self, message) -> dict:
+        """Convert standard format message to OpenAI format.
+
+        Converts structured content back to OpenAI's simple string format.
+
+        Args:
+            message: Message in standard format:
+                {
+                    "role": "user/assistant",
+                    "content": [{"type": "text", "text": str}]
+                }
+
+        Returns:
+            Message in OpenAI format:
+            {
+                "role": "user/assistant",
+                "content": str
+            }
+        """
+        # If content is already a string, return as-is
+        if isinstance(message.get("content"), str):
+            return message
+
+        # Convert structured content to string
+        if isinstance(message.get("content"), list):
+            text_parts = []
+            for part in message["content"]:
+                if part.get("type") == "text":
+                    text_parts.append(part["text"])
+
+            return {"role": message["role"], "content": " ".join(text_parts) if text_parts else ""}
+
         return message
 
-    # convert a message in this LLM's format to one or more messages in OpenAI format
     def to_standard_messages(self, obj) -> list:
+        """Convert OpenAI message to standard structured format.
```

Comment:
Same thing here. I don't think we want this change.

The approach I was taking with this is that everywhere in the whole Pipecat codebase wherever we deal with messages, we should always accept both:
-  the old-style OpenAI shortcut `"content": text` 
- the newer, more flexible `"content": List[ContentPart]`

This is kind of a pain. But there are too many corner cases in terms of where in code new messages might get defined and injected to do anything else, I think.

Reply:
If we want this, then the standard message format is both:
- the old-style OpenAI shortcut "content": text
- the newer, more flexible "content": List[ContentPart]

That means code needs to handle both formats, which is suboptimal. But, if it's not possible, then so be it. I'll revert this change and the from_standard_message format. That will require handling both old and new formats in the application code.

Reply:
Reverted + added docstring documenting behavior.